diff --git a/setup.py b/setup.py
index 8581a28..065678f 100644
--- a/setup.py
+++ b/setup.py
@@ -407,6 +407,7 @@ def customize_build_cf(EXTENSIONS, OPTIONS):
     del EXTENSIONS['jpeg12']
     del EXTENSIONS['jpegxl']
     del EXTENSIONS['zlibng']
+    del EXTENSIONS['blosc2']  # c-blosc2 library not commonly available
 
     # build the jpeg8 extension against libjpeg v9 instead of libjpeg-turbo
     EXTENSIONS['jpeg8']['cythonize'] = True
