diff --git a/setup.py b/setup.py
index 95fb75e..e1367cd 100644
--- a/setup.py
+++ b/setup.py
@@ -214,8 +214,8 @@ EXTENSIONS = {
     'ljpeg': ext(
         sources=['3rdparty/liblj92/lj92.c'], include_dirs=['3rdparty/liblj92']
     ),
-    'lz4': ext(libraries=['lz4']),
-    'lz4f': ext(libraries=['lz4']),
+    'lz4': ext(libraries=['liblz4']),
+    'lz4f': ext(libraries=['liblz4']),
     'lzf': ext(
         sources=['3rdparty/liblzf/lzf_c.c', '3rdparty/liblzf/lzf_d.c'],
         include_dirs=['3rdparty/liblzf'],
