From 8048cd5224a4433c1ba9d0f7b16f3a00cc677114 Mon Sep 17 00:00:00 2001
From: Serhii Kupriienko
Date: Wed, 4 Dec 2024 15:58:32 +0200
Subject: [PATCH] Fix default build

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index f0e4f7a..7cbe591 100644
--- a/setup.py
+++ b/setup.py
@@ -653,7 +653,7 @@ if 'sdist' not in sys.argv:
         elif os.environ.get('IMAGECODECS_CIBW', ''):
             customize_build = customize_build_cibuildwheel
         elif os.environ.get('CONDA_BUILD', ''):
-            customize_build = customize_build_condaforge
+            customize_build = customize_build_default
         elif shutil.which('port'):
             customize_build = customize_build_macports
         elif os.name == 'nt' and 'GCC' in sys.version:
-- 
2.47.0

